docs/proposals/resource-metrics-api.md

```diff
@@ -0,0 +1,177 @@
+available in API server so the use case initially won't be supported.
+
+#### Ubernetes
+
+Ubernetes might want to consider cluster-level usage (in addition to cluster-level request)
+of running pods when choosing where to schedule new pods. Although Ubernetes is still in design,
+we expect the metrics API described here to be sufficient. Cluster-level usage can be
+obtained by summing over usage of all nodes in the cluster.
+
+#### kubectl top
+
+This feature is not yet specified/implemented although it seems reasonable to support the following options:
+- ```kubectl top node <id>``` - list all pods running on the given node accompanied with theirs cpu/mem usage
+- ```kubectl top pod <id>``` - print cpu/mem usage stats for the given pod
+- ```kubectl top pod --selector=<label-query>``` - list all pods defined by the label query accompanied with their cpu/mem usage
+
+Since this feature has not been fully specified yet it will be not supported initally in the API although
+it will be probably possible to provide a reasonable implementation of the feature anyway.
+
+#### Kubernetes dashboard
+
+[Kubernetes dashboard](https://github.com/kubernetes/dashboard) in order to draw graps requires resource usage
+in timeseries format from relatively long period of time. The aggreations should be also possible on various levels
+including replication controllers, deployments, services, etc.
+
+Since the use case is complicated it will not be supported initally in the API and they will query Heapster
+directly using some custom API there.
+
+## Proposed API
+
+Initially the metrics API will be in a separate [API group](api-group.md) called ```metrics```.
+Later if we decided to have Node and Pod in different API groups also
+NodeMetrics and PodMetrics should be in different API groups.
+
+#### Schema
+
+The proposed schema is as follow. Each top-level object has `TypeMeta` and `ObjectMeta` fields
+to be compatible with Kubernetes API standards.
+
+```go
+type NodeMetrics struct {
+  unversioned.TypeMeta
+  ObjectMeta
+
+  // The following fields define time interval from which metrics were
+  // collected in the following format [Timestamp-Window, Timestamp].
+  Timestamp unversioned.Time
+  Window    unversioned.Duration
+
+  Usage v1.ResourceList
+}
+
+type PodMetrics struct {
+  unversioned.TypeMeta
+  ObjectMeta
+
+  // The following fields define time interval from which metrics were
+  // collected in the following format [Timestamp-Window, Timestamp].
+  Timestamp unversioned.Time
+  Window    unversioned.Duration
+
+  Containers []ContainerMetrics
+}
+
+type ContainerMetrics struct {
+  Name  string
+  Usage v1.ResourceList
+}
+```
+
+By default `Usage` is the mean from samples collected within the returned time window.
+The default time window is 1 minute.
+In the future we may want to introduce the following params:
+`aggreator` (`max`, `min`, `95th`, etc.) and `window` (`1h`, `1d`, `1w`, etc.)
```

Comment:
$0.02: instead of max/min/95th do p100/p0/p95. It's a little more consistent, and the "p" prefix suggests "percentile", while "th" only suggests some number in an ordering.

We should also include "avg".